src/background_script.js

```diff
@@ -6,20 +6,26 @@ const UNMONETIZED_ICON_PATH = '../assets/icon_unmonetized.png';
 // Listen to messages sent from Content Scripts via webBrowser.runtime.sendMessage
 webBrowser.runtime.onMessage.addListener(
 	(request, sender, sendResponse) => {
-		if (request.isCurrentlyMonetized) {
+		// Set browser badge and icon
+		if (request.isCurrentlyMonetized === true) {
 			webBrowser.browserAction.setBadgeBackgroundColor({ color: '#EF5E92' });
 			webBrowser.browserAction.setBadgeText({ text: '$' });
 			webBrowser.browserAction.setIcon({
 				path: MONETIZED_ICON_PATH,
 				tabId: sender.tab.id
 			});
-		} else {
+		}
+		if (request.isCurrentlyMonetized === false) {
```

Comment:
I like the idea, but it won't quite work because there are three cases rather than two: 
- request is not for the browser icon and does not contain `isCurrentlyMonetized`
- request is for the browser icon and `isCurrentlyMonetized` is true
- request is for the browser icon and `isCurrentlyMonetized` is false

I'll go with something similar though

Reply:
Went with sending `request.isCurrentlyMonetized.value` instead so that `background_script.js` knows the type of message being sent whether it is true or false. This is now consistent with how the messages from the `iframe_content.js` script work in `background_scipt.js`.

Reply:
If the request doesn't contain `isCurrentlyMonetized`, then `request.isCurrentlyMonetized` will be falsy, so wouldn't that be captured in the suggestion to define:

```
const isCurrentlyMonetized = ((request.isCurrentlyMonetized) && (request.isCurrentlyMonetized === true)) ? true : false;
```

Your solution works too (and lgtm!), but just want to confirm my understanding

Reply:
You're right that `request.isCurrentlyMonetized` will be falsy, but the request listener has to do nothing if `request.isCurrentlyMonetized` is `undefined`, do something if it is `true`, and another thing if it is `false`.  So just truthiness and falsiness is not enough, we need to specifically know if it is `true`, `false`, or `undefined`.

Reply:
I didn't pay enough attention to the code being called, I thought there were just two cases: true and false/undefined. Please disregard :)